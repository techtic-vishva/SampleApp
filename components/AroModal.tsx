import React from 'react'
import { View, StyleSheet, GestureResponderEvent } from 'react-native'
import { fill } from '../constants/Colors'
import { Text } from './Themed'
import OrangeButton from './OrangeButton'

export default function (props: {
	title: string
	cancelBtnText: string
	actionBtnTxt: string
	isCancleBtnDisable: boolean
	onActionPress?: (event: GestureResponderEvent) => void
	onCancelPress?: (event: GestureResponderEvent) => void
	tall?: boolean
}) {
	return (
		<>
			<View style={styles.mask}></View>
			<View style={[styles.popover, props.tall ? { height: 400, top: '25%' } : {}]}>
				<Text style={styles.titleStyle}>{props.title}</Text>
				<OrangeButton onPress={props.onActionPress} title={props.actionBtnTxt} outterStyle={styles.actionBtnStyle} />
				{!props.isCancleBtnDisable && (
					<OrangeButton onPress={props.onCancelPress} title={props.cancelBtnText} outterStyle={styles.cancelBtnStyle} />
				)}
			</View>
		</>
	)
}

const styles = StyleSheet.create({
	actionBtnStyle: {
		width: '75%',
		marginTop: 30
	},
	cancelBtnStyle: {
		width: '75%',
		marginTop: 10,
		backgroundColor: 'white',
		borderColor: 'black'
	},
	titleStyle: {
		color: 'black',
		fontFamily: 'objektiv-semi-bold',
		fontSize: 20,
		width: '75%',
		textAlign: 'center'
	},
	mask: {
		position: 'absolute',
		width: '100%',
		height: '100%',
		opacity: 0.6
	},
	popover: {
		position: 'absolute',
		top: '30%',
		width: '90%',
		height: 250,
		backgroundColor: 'white',
		borderRadius: 15,
		shadowColor: fill,
		display: 'flex',
		alignItems: 'center',
		justifyContent: 'center',
		alignSelf: 'center'
	}
})
