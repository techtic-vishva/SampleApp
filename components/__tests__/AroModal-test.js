import * as React from 'react'
import renderer from 'react-test-renderer'

import AroModal from '../AroModal'

it('render when cancel button disable properly', () => {
	const tree = renderer
		.create(
			<AroModal
				tile="Aro Model"
				cancelBtnText="cancel"
				actionBtnTxt="Delete"
				onActionPress={() => {
					console.log('Action button pressed')
				}}
				isCancleBtnDisable={true}
				onCancelPress={() => {}}
			/>
		)
		.toJSON()
	expect(tree).toMatchSnapshot()
})

it('render when cancel button show properly', () => {
	const tree = renderer
		.create(
			<AroModal
				tile="Aro Model"
				cancelBtnText="cancel"
				actionBtnTxt="nmhjkhkkDelete"
				onActionPress={() => {
					console.log('Action button pressed')
				}}
				isCancleBtnDisable={false}
				onCancelPress={() => {
					console.log('Cancel button pressed')
				}}
			/>
		)
		.toJSON()
	expect(tree).toMatchSnapshot()
})
