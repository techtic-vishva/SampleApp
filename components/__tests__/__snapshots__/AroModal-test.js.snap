// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render when cancel button disable properly 1`] = `
Array [
  <View
    style={
      Object {
        "height": "100%",
        "opacity": 0.6,
        "position": "absolute",
        "width": "100%",
      }
    }
  />,
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "white",
          "borderRadius": 15,
          "display": "flex",
          "height": 250,
          "justifyContent": "center",
          "position": "absolute",
          "shadowColor": "#23211e",
          "top": "30%",
          "width": "90%",
        },
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "objektiv",
          },
          Object {
            "color": "black",
            "fontFamily": "objektiv-semi-bold",
            "fontSize": 20,
            "textAlign": "center",
            "width": "75%",
          },
        ]
      }
    />
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ed832d",
          "borderColor": "#ed832d",
          "borderRadius": 10,
          "borderWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 30,
          "opacity": 1,
          "paddingHorizontal": 30,
          "paddingVertical": 12,
          "width": "75%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "objektiv",
            },
            Object {
              "color": "#0b0805",
              "fontSize": 18,
            },
          ]
        }
      >
        Delete
      </Text>
    </View>
  </View>,
]
`;

exports[`render when cancel button show properly 1`] = `
Array [
  <View
    style={
      Object {
        "height": "100%",
        "opacity": 0.6,
        "position": "absolute",
        "width": "100%",
      }
    }
  />,
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "white",
          "borderRadius": 15,
          "display": "flex",
          "height": 250,
          "justifyContent": "center",
          "position": "absolute",
          "shadowColor": "#23211e",
          "top": "30%",
          "width": "90%",
        },
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "objektiv",
          },
          Object {
            "color": "black",
            "fontFamily": "objektiv-semi-bold",
            "fontSize": 20,
            "textAlign": "center",
            "width": "75%",
          },
        ]
      }
    />
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ed832d",
          "borderColor": "#ed832d",
          "borderRadius": 10,
          "borderWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 30,
          "opacity": 1,
          "paddingHorizontal": 30,
          "paddingVertical": 12,
          "width": "75%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "objektiv",
            },
            Object {
              "color": "#0b0805",
              "fontSize": 18,
            },
          ]
        }
      >
        nmhjkhkkDelete
      </Text>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderColor": "black",
          "borderRadius": 10,
          "borderWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 10,
          "opacity": 1,
          "paddingHorizontal": 30,
          "paddingVertical": 12,
          "width": "75%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "objektiv",
            },
            Object {
              "color": "#0b0805",
              "fontSize": 18,
            },
          ]
        }
      >
        cancel
      </Text>
    </View>
  </View>,
]
`;
